bioconda_utils/lint_functions.py

```diff
@@ -185,6 +203,22 @@ def has_windows_bat_file(recipe, meta, df):
             'fix': 'remove windows .bat files'
         }
 
+
+def should_be_noarch(recipe, meta, df):
+    deps = _get_deps(meta)
+    if (
+        ('gcc' not in deps) and
+        ('python' in deps) and
+        not _has_preprocessing_selector(meta)
```

Comment:
Sorry didn't get a chance to reply before -- If there's a pre-processing selector in the url or test section, doesn't that mean it's arch-specific?

Either way I think it's ok at least for now to let some packages slide until we get a better feel for what works with noarch, and then tighten up the linter later.